runtime/bcutil/ROMClassCreationContext.hpp

```diff
@@ -425,9 +426,26 @@ class ROMClassCreationContext
 						/* for hidden class className has ROM address appended at the end, _className does not have that */
 						classNameLenToCompare1 = (U_16)_classNameLength;
 					}
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+#define J9_INJECTED_INVOKER "InjectedInvoker/0x0"
+					if (0 == memcmp(className, J9_INJECTED_INVOKER, sizeof(J9_INJECTED_INVOKER) - 1)) {
+#undef J9_INJECTED_INVOKER
+						/* className has InjectedInvoker/0x0000000000000000. */
+#define J9_INJECTED_INVOKER "$$InjectedInvoker"
+						U_8 *start = _className + _classNameLength - (sizeof(J9_INJECTED_INVOKER) - 1);
+						if (0 == memcmp(start, J9_INJECTED_INVOKER, sizeof(J9_INJECTED_INVOKER) - 1)) {
```

Comment:
also just to be clear, is this the format?

className is always `InjectedInvoker/0x0000000000000000` 
_className is always `[someClassName]$$InjectedInvoker`

Reply:
yes

Reply:
I wasn't sure if `0x0000000000000000` will always have a constant number of zeros. So, I chose to partially match it: `0x0`.